bot/exts/info/information.py

```diff
@@ -541,13 +544,14 @@ async def rules(self, ctx: Context, *args: Optional[str]) -> Optional[Set[int]]:
             for rule_keyword in rule_keywords:
                 keyword_to_rule_number[rule_keyword] = rule_number
 
-        for word in args:
-            try:
-                rule_numbers.append(int(word))
-            except ValueError:
-                if (kw := word.lower()) not in keyword_to_rule_number:
-                    break
-                keywords.append(kw)
+        if args:
+            for word in args.split():
+                try:
+                    rule_numbers.append(int(word))
+                except ValueError:
+                    if (kw := word.lower()) not in keyword_to_rule_number:
+                        break
+                    keywords.append(kw)
```

Comment:
I know you didn't add this code in this PR, but it feel wrong? Surely we don't want to break and stop processing the rest of the args if just one of them doesn't match. I think this would be better, so even if one doens't match, we still grab out the rest.
```suggestion
                    if (kw := word.lower()) in keyword_to_rule_number:
                        keywords.append(kw)
```

Reply:
I'm thinking maybe it's to do with time spent in checking messages if someone decides to do a `rules` with 4000 chars or something? Would it be more sensible to impose a parsing limit of, say 50 words?